modules/core/src/main/scala/ch/epfl/bluebrain/nexus/admin/core/resources/Resources.scala

```diff
@@ -62,37 +66,37 @@ class Resources[F[_], A: IdResolvable: TypeFilterExpr](agg: Agg[F], sparqlClient
     * @return () or the appropriate rejection in the ''F'' context
     */
   @silent
-  def validateUpdate(id: A, value: Json): F[Unit] = F.pure(())
+  def validateUpdate(id: A, value: Json): F[Unit]
+
+  def tags: Set[String]
 
   /**
     * Asserts the resource exists and it allows modifications on children resources.
     *
-    * @param persId the unique persistence ID of the resource
+    * @param id  the unique id of the resource
     * @return () or the appropriate rejection in the ''F'' context
     */
-  def validateUnlocked(persId: String): F[Unit] =
-    agg.currentState(persId) flatMap {
-      case Initial                   => F.raiseError(CommandRejected(ParentResourceDoesNotExists))
-      case Current(_, _, _, _, true) => F.raiseError(CommandRejected(ResourceIsDeprecated))
-      case _                         => F.pure(())
+  def validateUnlocked(id: A): F[Unit] =
+    agg.currentState(id.persistenceId) flatMap {
+      case Initial                      => F.raiseError(CommandRejected(ParentResourceDoesNotExist))
+      case Current(_, _, _, _, _, true) => F.raiseError(CommandRejected(ResourceIsDeprecated))
+      case _                            => F.pure(())
     }
 
   /**
     * Attempts to create a new resource instance.
     *
     * @param id     the identifier of the resource
     * @param value  the json payload of the resource
-    * @param tags   the tags added to the consequent [[ResourceEvent]] which might be created as a result of this operation
-    * @param persId the unique persistence ID of the resource
     * @return a [[RefVersioned]] instance wrapped in the abstract ''F[_]'' type
     *         if successful, or a [[ch.epfl.bluebrain.nexus.admin.core.Fault]] wrapped within ''F[_]'' otherwise
     */
-  def create(id: A, value: Json)(tags: Set[String], persId: String)(
-      implicit ctx: CallerCtx,
-      @silent perms: HasCreateProjects): F[RefVersioned[A]] =
+  def create(id: A, value: Json)(implicit ctx: CallerCtx): F[RefVersioned[A]] =
```

Comment:
you need an evidence that you can call that method, otherwise anybody can.
`@silent perms: HasCreateProjects`

Reply:
I think it should be handled at `Route` level. Even if we have `perms: HasCreateProjects`, it doesn't mean that it is on this project/org, e.g. user could have `projects/create` on `A` org and try to create project in `B`  and we still have to check for that. We already do that on `Route` level, so I don't think we need to do it here again.

Reply:
If it is only handled in the routes, nothing prevent you from calling this method form the routes or from anywhere else. This way of programming is a safeguard, used for example by [slick](https://danielwestheide.com/blog/2015/06/28/put-your-writes-where-your-master-is-compile-time-restriction-of-slick-effect-types.html)

Reply:
But the compile type guarantee in this case only gives us the information that the user has a given permission somewhere, not necessarily on the resource they are trying to create/update/fetch, so I don't think it's very useful here, unless we can define it in a way that it restricts the permissions to a certain resource path.